lib/datadog/appsec/assets/waf_rules/recommended.json

```diff
@@ -1,9 +1,34 @@
 {
   "version": "2.2",
   "metadata": {
-    "rules_version": "1.3.1"
+    "rules_version": "1.4.1"
   },
   "rules": [
```

Comment:
`libddwaf` being truly just an engine that is fairly oblivious to the rules (except their format itself), it's quite independent of the ruleset. Notably, vendoring the ruleset into `libddwaf-rb` creates trickiness, e.g how to bump versions of that gem.

What I did consider though is packaging these in a separate gem and leveraging semver, so that it can be updated without doing `ddtrace` releases.

Reply:
> libddwaf being truly just an engine that is fairly oblivious to the rules

Yes and no. In the strictest sense, yes, someone could be using libddwaf-rb for something else other than dd-trace-rb. But... how often would that be the case? I would bet very strongly that would never, ever happen. And if it did, well, customers could open a ticket and we could decide to change our minds ;)

> Notably, vendoring the ruleset into libddwaf-rb creates trickiness, e.g how to bump versions of that gem.

Well, doesn't libddwaf-rb versioning already consider these things? E.g., if we update the rules, we release `1.5.1.0.1`, then `1.5.1.0.2`...

But! This was just a suggestion, so if you're not a big fan, we can discuss it later/afterwards/never ;)

Reply:
> rulesets and libddwaf are extremely independent

I... somewhat disagree. Or at least, at the level we're talking about. At the level of "these are two black-box components that appsec needs to work", they are actually quite coupled. Either is quite useless without the other for 99.999999999% of our customers.

The way I see it, they are both conceptually a combined "lib-this-is-needed-for-appsec-to-actually-work".

Having rulesets inside libbdwaf I believe would enable faster releases -- because we wouldn't need to do the full QA for dd-trace-rb to a release.

I agree that all my arguments would apply as well if the rulesets were in a standalone gem, but to be honest, it still seems to me that that's exposing an internal implementation seam ("we think of them as separate") to end-users, as again, I don't expect anyone to ever reasonably separate the two. It's a bit like if I was to separate different folders of the profiler into gems. Yes, they are independent components but nobody is going to use them as so 😉 😉 😉

Reply:
I can understand that under a technical angle, but from a user's perspective the ruleset is more like a configurable input to ddtrace, the latter being a black box that happens to pull in libddwaf as a dependency. In some way, the user doesn't care at all about libddwaf but cares _a lot_ about the ruleset.

Indeed the ruleset can be tweaked, tuned, downloaded, and fed via configuration to ddtrace, entirely by the user, overriding the default that we provide. In fact, the vendored ruleset is a sort of already-primed cache, provided as a works-out-of-the-box courtesy since in theory such ruleset vendoring would be unneeded if ddtace were able to fetch it from the backend directly (hint, hint, remote config).

Reply:
(I believe we can pursue this discussion internally if you so desire :)